addons/google_calendar/models/google_sync.py

```diff
@@ -176,8 +177,10 @@ def _sync_google2odoo(self, google_events: GoogleEvent, default_reminders=()):
             # This could be dangerous if google server time and odoo server time are different
             updated = parse(gevent.updated)
             odoo_record = self.browse(gevent.odoo_id(self.env))
+            # Use the record's write_date to apply Google updates only if they are newer than Odoo's write_date.
+            odoo_record_write_date = write_dates[(self._name, odoo_record.id)] if (self._name, odoo_record.id) in write_dates else odoo_record.write_date
```

Comment:
Could be simplified by splitting the context into 2.
```suggestion
            odoo_record_write_date = write_dates.get(odoo_record.id, odoo_record.write_date)
```